keylime/payloads/keylime-controller/local_action_node_failover.py

```diff
@@ -1,27 +1,23 @@
 import os
 import asyncio
+import subprocess
 
-async def execute(revocation):
-    """Node fail over script
+"""Node fail over script
 
-    This script will be sent to the keylime_agent running on a machine
-    which has an OpenShift client (oc) installed and configured.
+This script will be sent to the keylime_agent running on a machine
+which has an OpenShift client (oc) installed and configured.
 
-    When a node is compromised this script will run `oc adm cordon` and
-    `oc adm drain` commands to remove that worker node from the OpenShift
-    cluster
-    """
-    if revocation['type']!='revocation':
-        return
+When a node is compromised this script will run `oc adm cordon` and
+`oc adm drain` commands to remove that worker node from the OpenShift
+cluster
+"""
 
-    # TODO - get the node name
-    node_name = "XXX"
-    rc = os.system("oc adm cordon {}".format(node_name))
-    if rc != 0:
-        print("Could not cordon node {}: Command failed with code {}".format(node_name, rc))
+async def execute(revocation):
+    if revocation['type']!='revocation':
         return
 
-    rc = os.system("oc adm drain {}".format(node_name))
-    if rc != 0:
-        print("Could not drain node {}: Command failed with code {}".format(node_name, rc))
-        return
+    node_name = 'mpeters-okd-zxrh5-' + revocation['agent_id']
```

Comment:
Instead of worrying about node names, we could just use a node selector.

Reply:
Yes, this is definitely a fragile way of doing it and something I want to make more robust later.